plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -210,10 +210,7 @@ private FileWriter createParquetFileWriter(FileSystem fileSystem, Path path, Lis
         CompressionCodecName compressionCodecName = getCompressionCodec(session).getParquetCompressionCodec();
 
         try {
-            Callable<Void> rollbackAction = () -> {
-                fileSystem.delete(path, false);
-                return null;
-            };
+            Closeable rollbackAction = () -> fileSystem.delete(path, false);
```

Comment:
To me making the rollback action a `Closeable` looks weird, there is no resource that we are releasing.
You could use `Runnable` instead if the problem is that you want to avoid extra `return null`.

Reply:
> You could use Runnable instead if the problem is that you want to avoid extra return null.

I actually use it in `try-with-resource`, e.g:
```
        try (rollbackAction) {
            avroWriter.close();
        }
        catch (Exception e) {
            throw new TrinoException(ICEBERG_WRITER_CLOSE_ERROR, "Error rolling back write to Avro file", e);
        }
```
so I don't have to deal with syntax sugar myself. I think using native Java `try-with-resource` is actually better than relying on Guava closer.